docs/3.registering-abilities.md

```diff
@@ -0,0 +1,276 @@
+### 3. Registering Abilities (`wp_register_ability`)
+
+The primary way to add functionality to the Abilities API is by using the `wp_register_ability()` function, typically hooked into the `abilities_api_init` action.
+
+**Function Signature**
+
+```php
+wp_register_ability( string $id, array $args ): ?\WP_Ability
+```
+
+-   `$id` (`string`): A unique identifier for the ability.
+-   `$args` (`array`): An array of arguments defining the ability configuration.
+-   **Return:** (`?\WP_Ability`) An instance of the registered ability if the it was successfully registered, `null` on failure (e.g., invalid arguments, duplicate ID).
+
+**Parameters Explained**
+
+The `$args` array accepts the following keys:
+
+-   `label` (`string`, **Required**): A human-readable name for the ability. Used for display purposes. Should be translatable.
+-   `description` (`string`, **Required**): A detailed description of what the ability does, its purpose, and its parameters or return values. This is crucial for AI agents to understand how and when to use the ability. Should be translatable.
+-   `input_schema` (`array`, **Required**): A JSON Schema definition describing the expected input parameters for the ability's execute callback. Used for validation and documentation.
+-   `output_schema` (`array`, **Required**): A JSON Schema definition describing the expected format of the data returned by the ability. Used for validation and documentation.
+-   `execute_callback` (`callable`, **Required**): The PHP function or method to execute when this ability is called.
+    -   The callback receives one argument: an associative array of validated input parameters.
+    -   The callback should return the result of the ability's operation or throw an exception on failure.
+-   `permission_callback` (`callable`|`null`, **Optional**): A callback function to check if the current user has permission to execute this ability.
+    -   Should return `true` if the user has permission, `false` or a `WP_Error` object otherwise.
+    -   If not provided, the ability will only validate input parameters before execution.
+    -   Use `'__return_true'` for publicly accessible abilities.
+-   `meta` (`array`, **Optional**): An associative array for storing arbitrary additional metadata about the ability.
+
+**Ability ID Convention**
+
+The `$id` parameter must follow the pattern `namespace/ability-name`:
+-   **Format:** Must contain only lowercase alphanumeric characters (`a-z`, `0-9`), hyphens (`-`), and one forward slash (`/`) for namespacing.
+-   **Convention:** Use your plugin slug as the namespace, like `my-plugin/ability-name` or `woocommerce/get-product`.
+-   **Examples:** `my-plugin/update-settings`, `contact-form/send-message`, `analytics/track-event`
```

Comment:
Nit, since there are examples listed, maybe this would work better:
```suggestion
-   **Convention:** Use your plugin slug as the namespace, like `my-plugin/ability-name`.
-   **Examples:** `my-plugin/update-settings`, `woocommerce/get-product`, `contact-form/send-message`, `analytics/track-event`
```